app/models/supplejack_api/collection_metric.rb

```diff
@@ -31,27 +33,54 @@ class CollectionMetric
 
     def self.spawn
       return unless SupplejackApi.config.log_metrics == true
-      dates = SupplejackApi::RecordMetric.all.map(&:date).uniq
-
+      dates = SupplejackApi::RecordMetric.where(:date.lt => Time.zone.now.beginning_of_day).map(&:date).uniq
       dates.each do |date|
         collections = SupplejackApi::RecordMetric.where(date: date).map(&:display_collection).uniq
 
         collections.each do |collection|
-          find_or_create_by(date: date, display_collection: collection).inc(
-            searches: record_metric(date, collection).sum(:appeared_in_searches),
-            record_page_views: record_metric(date, collection).sum(:page_views),
-            user_set_views: record_metric(date, collection).sum(:user_set_views),
-            user_story_views: record_metric(date, collection).sum(:user_story_views),
-            records_added_to_user_sets: record_metric(date, collection).sum(:added_to_user_sets),
-            records_added_to_user_stories: record_metric(date, collection).sum(:added_to_user_stories),
-            total_source_clickthroughs: record_metric(date, collection).sum(:source_clickthroughs)
-          ).save!
+          record_metrics = record_metrics_to_be_processed(date, collection)
+          collection_metrics = find_or_create_by(date: date, display_collection: collection).inc(
+            searches: record_metrics.sum(:appeared_in_searches),
```

Comment:
this increments these fields by the values provided right?

Reply:
Correct